File: src/com/intellij/idea/plugin/hybris/flexibleSearch/highlighting/FlexibleSearchHighlighterColors.java
/*
 * This file is part of "hybris integration" plugin for Intellij IDEA.
 * Copyright (C) 2014-2016 Alexander Bartash <AlexanderBartash@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

package com.intellij.idea.plugin.hybris.flexibleSearch.highlighting;

import com.intellij.openapi.editor.colors.TextAttributesKey;
import com.intellij.openapi.editor.markup.TextAttributes;
import com.intellij.ui.JBColor;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

import java.awt.*;

import static com.intellij.ide.highlighter.JavaHighlightingColors.CLASS_NAME_ATTRIBUTES;
import static com.intellij.openapi.editor.DefaultLanguageHighlighterColors.BRACES;
import static com.intellij.openapi.editor.DefaultLanguageHighlighterColors.BRACKETS;
import static com.intellij.openapi.editor.DefaultLanguageHighlighterColors.INSTANCE_FIELD;
import static com.intellij.openapi.editor.DefaultLanguageHighlighterColors.KEYWORD;
import static com.intellij.openapi.editor.DefaultLanguageHighlighterColors.LINE_COMMENT;
import static com.intellij.openapi.editor.DefaultLanguageHighlighterColors.NUMBER;
import static com.intellij.openapi.editor.DefaultLanguageHighlighterColors.PARENTHESES;
import static com.intellij.openapi.editor.DefaultLanguageHighlighterColors.STRING;

public final class FlexibleSearchHighlighterColors {

    private FlexibleSearchHighlighterColors() {
    }

    public static final TextAttributesKey FS_KEYWORD = key("KEYWORD", KEYWORD);
    public static final TextAttributesKey FS_STRING = key("STRING", STRING);

    public static final TextAttributesKey FS_BRACES = key("BRACES", BRACES);
    public static final TextAttributesKey FS_PARENTHESES = key("PARENTHESES", PARENTHESES);
    public static final TextAttributesKey FS_BRACKETS = key("BRACKETS", BRACKETS);

    public static final TextAttributesKey FS_SYMBOL = key("COMMA", KEYWORD);
    public static final TextAttributesKey FS_NUMBER = key("NUMBER", NUMBER);

    public static final TextAttributesKey FS_COLUMN = key("COLUMN", INSTANCE_FIELD);
    public static final TextAttributesKey FS_TABLE = key("TABLE NAME", CLASS_NAME_ATTRIBUTES);

    public static final TextAttributesKey FS_COMMENT = key("COMMENT", LINE_COMMENT);

    public static final TextAttributesKey FS_PARAMETER =
        TextAttributesKey.createTextAttributesKey("FS_PARAMETER", new TextAttributes(
            new JBColor(0x0D96D9, 0x0097C5),
            null,
            null,
            null,
            Font.PLAIN
        ));


    private static TextAttributesKey key(
        @NonNls @NotNull final String externalName,
        final TextAttributesKey fallbackAttributeKey
    ) {
        return TextAttributesKey.createTextAttributesKey(externalName, fallbackAttributeKey);
    }
}

File: src/com/intellij/idea/plugin/hybris/flexibleSearch/highlighting/FlexibleSearchHighlighterColors.kt
/*
 * This file is part of "SAP Commerce Developers Toolset" plugin for IntelliJ IDEA.
 * Copyright (C) 2014-2016 Alexander Bartash <AlexanderBartash@gmail.com>
 * Copyright (C) 2019-2024 EPAM Systems <hybrisideaplugin@epam.com> and contributors
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.intellij.idea.plugin.hybris.flexibleSearch.highlighting

import com.intellij.openapi.editor.DefaultLanguageHighlighterColors
import com.intellij.openapi.editor.colors.TextAttributesKey

object FlexibleSearchHighlighterColors {

    val FXS_KEYWORD = key("FXS_KEYWORD", DefaultLanguageHighlighterColors.KEYWORD)
    val FXS_STRING = key("FXS_STRING", DefaultLanguageHighlighterColors.STRING)
    val FXS_BRACES = key("FXS_BRACES", DefaultLanguageHighlighterColors.BRACES)
    val FXS_DBRACES = key("FXS_DBRACES", DefaultLanguageHighlighterColors.BRACES)
    val FXS_PARENS = key("FXS_PARENS", DefaultLanguageHighlighterColors.PARENTHESES)
    val FXS_BRACKETS = key("FXS_BRACKETS", DefaultLanguageHighlighterColors.BRACKETS)
    val FXS_SYMBOL = key("FXS_SYMBOL", DefaultLanguageHighlighterColors.KEYWORD)
    val FXS_NUMBER = key("FXS_NUMBER", DefaultLanguageHighlighterColors.NUMBER)
    val FXS_COLUMN = key("FXS_COLUMN", DefaultLanguageHighlighterColors.INSTANCE_FIELD)
    val FXS_TABLE = key("FXS_TABLE_NAME", DefaultLanguageHighlighterColors.FUNCTION_DECLARATION)
    val FXS_COMMENT = key("FXS_COMMENT", DefaultLanguageHighlighterColors.LINE_COMMENT)
    val FXS_FUNCTION_CALL = key("FXS_FUNCTION_CALL", DefaultLanguageHighlighterColors.STATIC_METHOD)
    val FXS_OUTER_JOIN = key("FXS_OUTER_JOIN", DefaultLanguageHighlighterColors.INLINE_PARAMETER_HINT_HIGHLIGHTED)
    val FXS_LOCALIZED = key("FXS_LOCALIZED", DefaultLanguageHighlighterColors.INLINE_PARAMETER_HINT)
    val FXS_PARAMETER = key("FXS_PARAMETER", DefaultLanguageHighlighterColors.PARAMETER)
    val FXS_COLUMN_SEPARATOR = key("FXS_COLUMN_SEPARATOR", DefaultLanguageHighlighterColors.DOT)
    val FXS_TABLE_ALIAS = key("FXS_TABLE_ALIAS", DefaultLanguageHighlighterColors.STATIC_FIELD)
    val FXS_COLUMN_ALIAS = key("FXS_COLUMN_ALIAS", DefaultLanguageHighlighterColors.STATIC_FIELD)
    val FXS_TABLE_TAIL = key("FXS_TABLE_TAIL", DefaultLanguageHighlighterColors.PREDEFINED_SYMBOL)
    val FXS_BOOLEAN = key("FXS_BOOLEAN", DefaultLanguageHighlighterColors.KEYWORD)
    val FXS_STAR = key("FXS_STAR", DefaultLanguageHighlighterColors.STATIC_FIELD)
    val FXS_OPERATION_SIGN = key("FXS_OPERATION_SIGN", DefaultLanguageHighlighterColors.OPERATION_SIGN)

    private fun key(
        externalName: String,
        fallbackAttributeKey: TextAttributesKey
    ) = TextAttributesKey.createTextAttributesKey(externalName, fallbackAttributeKey)
}